libs/SmartSync/SmartSync/Classes/Manager/SFSmartSyncSyncManager.m

```diff
@@ -570,36 +482,32 @@ - (void)syncUpOneEntry:(SFSyncState*)sync
      * Checks if we are attempting to update a record that has been updated
      * on the server AFTER the client's last sync down. If the merge mode
      * passed in tells us to leave the record alone under these
-     * circumstances, we will do nothing and return here.
+     * circumstances, we will do nothing.
      */
     if (mergeMode == SFSyncStateMergeModeLeaveIfChanged && !locallyCreated) {
         // Need to check the modification date on the server, against the local date.
         __weak typeof(self) weakSelf = self;
-        SFSyncUpRecordModificationResultBlock modificationBlock = ^(NSDate *localDate, NSDate *serverDate, NSError *error) {
+        [target isNewerThanServer:self record:record resultBlock:^(BOOL isNewerThanServer) {
```

Comment:
Determining if the server data changed can be more involved than fetching a last modified date (e.g. in the case of record trees) - so letting the target do the work.